scripts/us_census/pep/monthly_population_estimate/preprocess.py

```diff
@@ -378,30 +408,269 @@ def process(self):
         calls defined methods to clean, generate final
         cleaned CSV file, MCF file and TMCF file.
         """
-        for file in self._input_files:
+        #input_path = FLAGS.input_path
+        ip_files = os.listdir(self.input_path)
+        self.input_files = [
+            self.input_path + os.sep + file for file in ip_files
+        ]
+        processed_count = 0
+        total_files_to_process = len(self.input_files)
+        logging.info(f"No of files to be processed {len(self.input_files)}")
+        for file in self.input_files:
             df = self._load_data(file)
-            self._transform_data(df)
-        self._generate_mcf(self._df.columns)
-        self._generate_tmcf(self._df.columns)
+            result = self._transform_data(df)
+            if result:
+                processed_count += 1
+            else:
+                logging.fatal(f'Failed to process {file}')
+        logging.info(f"No of files processed {processed_count}")
+        if processed_count == total_files_to_process & total_files_to_process > 0:
+            self._generate_mcf(self._df.columns)
+            self._generate_tmcf(self._df.columns)
+        else:
+            logging.fatal(
+                "Aborting output files as no of files to process not matching processed files"
+            )
 
 
-def main(_):
-    input_path = FLAGS.input_path
+def add_future_year_urls():
+    """
+    This method scans the download URLs for future years.
+    """
+    global _FILES_TO_DOWNLOAD
+    with open(os.path.join(_MODULE_DIR, 'input_url.json'), 'r') as inpit_file:
+        _FILES_TO_DOWNLOAD = json.load(inpit_file)
+    urls_to_scan = [
+        "https://www2.census.gov/programs-surveys/popest/tables/2020-{YEAR}/national/totals/NA-EST{YEAR}-POP.xlsx"
+    ]
+    # This method will generate URLs for the years 2021 to 2029
+    for future_year in range(2021, 2029):

```

Comment:
Improved sentence:

The current for loop seems to be adding URLs for multiple years, while the requirement is to add only one future year's URL based on the `input_url.json` file. 

Please modify the code as follows:
1. Construct the URL for the current year.
2. Check if the constructed URL exists:
   - If yes, add it to the `_FILES_TO_DOWNLOAD` list.
   - If not, construct the URL for the previous year and check its existence. If found, add it to the `_FILES_TO_DOWNLOAD` list.

Reply:
Changes modified @krishnaswamypradeep